app/controllers/dor_controller.rb

```diff
@@ -47,24 +47,36 @@ def label
     respond_with params.merge('label' => Dor::MetadataService.label_for(params[:source_id]))
   end
 
+  ##
+  # reindex will make a synchronous index update to the Solr core
+  #
+  # @option params [String] `:pid` the druid for the object to reindex
+  # @option params [String] `:asynchronous` if present does an asynchronous transaction with solr index (i.e., no commit)
   def reindex
-    obj = Dor.load_instance params[:pid]
+    obj = Dor.load_instance params[:pid] # could be any type of Dor object
```

Comment:
here's the part in the other branch that seems relevant here:
https://github.com/sul-dlss/argo/pull/246/files#diff-881a03fbeec456e041aa84741b3b7dd9R45
https://github.com/sul-dlss/argo/pull/246/files#diff-881a03fbeec456e041aa84741b3b7dd9R7
https://github.com/sul-dlss/argo/pull/246/files#diff-44392392ca7d32d0aa276eac0fa5e7a8R32
https://github.com/sul-dlss/argo/pull/246/files#diff-44392392ca7d32d0aa276eac0fa5e7a8R7